app/javascript/spec/service-form/service-form.spec.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import fetchMock from 'fetch-mock';
+import FormRender from '@data-driven-forms/react-form-renderer';
+import ServiceForm from '../../components/service-form';
+
+describe('Cloud tenant form component', () => {
+  let initialProps;
+  let submitSpy;
+  let flashSpy;
+
+  beforeEach(() => {
+    initialProps = {
+      maxNameLen: 1,
```

Comment:
Submit can't ever be undisabled when validation fails, right?

This needs to be at least `"foo".length`